release-notes/7.0/supported-os.json

```diff
@@ -184,6 +184,12 @@
                     ],
                     "unsupported-versions": [
                         "10"
+                    ],
+                    "exceptional-versions": [
+                        {
+                            "version": "11",
+                            "note": "Debian 11 is continued to be supported after EOL as it's used as [container base](https://github.com/dotnet/dotnet-docker/discussions/5710)."
+                        }
```

Comment:
@richlander This is based on my assumption/understanding that the container image was based off from Debian 11 similar to .NET 6, so this would also have been true at EOL state.

Reply:
This one looks good for 6.0 but not 7.0. 7.0 was already EOL when Debian 11 went EOL.

- https://wiki.debian.org/DebianReleases
- https://github.com/dotnet/core/blob/main/releases.md

Reply:
Ok, so the images for .NET 7 were based on Debian 12 then? Thought I saw 11 somewhere.

Reply:
They were based on Debian 11, but since Debian 11 went EOL after .NET 7, this exception isn't important.

Reply:
I see the field more as "noteworthy version" with added information about what makes this special, not just about "this has an effect on the lifecycle". Therefore, I would expect this to been there right from the get-go of the .NET version, not just when Debian 11 went EOL. Same reason why I added it to .NET 8 while Debian 12 is still in support.

Reply:
This is the same case as https://github.com/dotnet/core/pull/9481#issuecomment-2307256100. 

When .NET 7 went EOL, Debian 11 was still supported by Debian maintainers. That's the final state of .NET 7 support, so no extra information is needed.

The tool should not check if "supported" OSes are actually still supported for EOL .NET versions.

Reply:
This comment is IMO not about a state that applied AFTER EOL but applied to that version even starting with 8.0.0. See https://github.com/Falco20019/core/blob/e3714e897be3ad3347ce635fc0056d8f96ffc41d/release-notes/8.0/supported-os.json#L162-L167 where we also show Debian 12 (which is currently Active, not EOL). The one in .NET 7 is also not for checking "supported" OSes (as .NET 7 is EOL itself) but about reflecting the file as it SHOULD have been when .NET 7 became EOL.

But I have no strong feelings on that, we can also just keep it as is.

Reply:
Sorry ... I've been travelling so short on time.

The 8.0 example/change makes good sense. Fully on board.

It's the .NET 7 one that is different since Debian 11 was not supported after it's EOL since .NET 7 EOL came first.

Reply:
Would be good to get more opinions on this. But since there aren't really users besides us, I think there won't be too much input soon. So I'm fine either way for the moment as long as we have the exceptions stated latest when something is EOL. Me personally, I don't know about all the things all the time, so having them upfront would help me. But since I'm not running into issues, it's fine for me either way. I just wouldn't be able to add them :)